include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -573,6 +574,48 @@ class __future : private std::tuple<_Args...>
     }
 };
 
+// A contract for a future class for reduce: <execution policy, sycl::event, USM host memory for the reduced value>
+template <typename _ExecutionPolicy, typename _Event, typename _Res>
+class __reduce_future
+{
+    _ExecutionPolicy __my_exec;
+    _Event __my_event;
+    ::std::shared_ptr<_Res> __my_res;
```

Comment:
Should this class be copy constructible or should we allow shared ownership? If we do not need it to be copy constructible, then I think the best option is to delete the copy constructor / assignment operator and make this `::std::unique_ptr<_Res>`

Reply:
I tend to agree that our future-like objects should not be copied similar to `std::future`. I will change that if there are no other opinions on when copying a future could be useful.

Reply:
Fixed: now we using ::std::unique_ptr<_Res> with custom deleter.